spec/components/candidate_interface/apply_again_add_another_course_component_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::ApplyAgainAddAnotherCourseComponent do
+  before do
+    FeatureFlag.activate(:apply_again_with_three_choices)
+  end
+
+  let(:application_form) { create(:application_form, phase: 'apply_2') }
+
+  context 'Add another course button in apply again' do
+    it 'renders you can add 2 more courses if the number of application choices is 1' do
+      create(:application_choice, application_form: application_form)
+      result = render_inline(described_class.new(application_form: application_form))
+      expect(result.text).to include('You can add 2 more courses')
+    end
+
+    it 'renders you can add 1 more courses if the number of application choices is 2' do
+      create(:application_choice, application_form: application_form)
+      create(:application_choice, application_form: application_form)
+      result = render_inline(described_class.new(application_form: application_form))
```

Comment:
You can move the result to the subject of the spec and separate the specs into contexts. 

e.g

```ruby
  subject(:result) { render_inline(described_class.new(application_form: application_form)) }

  context 'Add another course button in apply again' do
    context 'when the number of courses is 1' do
      before do
        create(:application_choice, application_form: application_form)
      end
    end

    context 'when the number of courses is 2' do
      before do
        create(:application_choice, application_form: application_form)
        create(:application_choice, application_form: application_form)
      end
    end
  end
```